### DISCUSSION TITLE
Understanding the Figma and Storybook workflow

### DISCUSSION THREAD

**Discussion by nicolaspiquita** [storybookjs/storybook#24875] [Help] [answered]
Hi everyone, can someone help me understand the relationship between Figma and Storybook? What I understand would make sense is that you have your UI components in Figma and when you create and update the components in Figma and publish them to the library, it should update Storybook and your React app (after approval of course) so that there is a single source of truth and easy to view the latest components and updates made in Figma.

I don't see how to do this if it is possible. I have experience with and understand how a React app and Storybook work together, but I am a UI/UX designer and don't understand how Figma fits in and how the Chromatic plugin works with this workflow. It still feels as if Figma is a completely separate process and that whatever you do in Figma, you need to manually implement the Figma changes in your React App and then push to Storybook.

Hope this makes sense and appreciate any help/input.

**Comment by joevaugh4n** [comment, marked as answer]
Hey @nicolaspiquita! Your understanding is correct. Designing in Figma _is_ a separate process to developing the actual coded UI inside Storybook. However, the plugin helps you by letting you view your Figma designs and your coded UI (in Storybook) side-by-side. That makes the process of aligning design and dev much simpler, because you don't need to constantly switch between different tabs to check that your implementation matches the designs!

https://www.youtube.com/watch?v=O3afqyCS2-o&t=384s

Happy to chat through this further if you have any other questions!

**Reply by nicolaspiquita** [reply]
Thanks @joevaugh4n this helps better understand the process. Pity we can't work within Figma, then the design work is automatically pushed to the React project and Storybook instead of having to work in Figma and then have to go and code the design work in VS code thereafter. Maybe Figma's dev mode will mature to a point where you set all the properties within Figma dev mode, and then it gets pushed to the code base and Storybook gets updated based on this or with a single terminal command.
